com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/AssetPostProcessors/MaterialPostprocessor.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Rendering.BuiltIn;
+using UnityEngine;
+using static Unity.Rendering.BuiltIn.ShaderUtils;
+
+namespace UnityEditor.Rendering.BuiltIn
+{
+    class MaterialModificationProcessor : AssetModificationProcessor
+    {
+        static void OnWillCreateAsset(string asset)
+        {
+            if (!asset.ToLowerInvariant().EndsWith(".mat"))
+            {
+                return;
+            }
+            MaterialPostprocessor.s_CreatedAssets.Add(asset);
+        }
+    }
+
+    class MaterialReimporter : Editor
+    {
+        static bool s_NeedToCheckProjSettingExistence = true;
+
+        static void ReimportAllMaterials()
+        {
+            string[] guids = AssetDatabase.FindAssets("t:material", null);
+            // There can be several materials subAssets per guid ( ie : FBX files ), remove duplicate guids.
+            var distinctGuids = guids.Distinct();
+
+            int materialIdx = 0;
+            int totalMaterials = distinctGuids.Count();
+            foreach (var asset in distinctGuids)
+            {
+                materialIdx++;
+                var path = AssetDatabase.GUIDToAssetPath(asset);
+                EditorUtility.DisplayProgressBar("Material Upgrader re-import", string.Format("({0} of {1}) {2}", materialIdx, totalMaterials, path), (float)materialIdx / (float)totalMaterials);
+                AssetDatabase.ImportAsset(path);
+            }
+            EditorUtility.ClearProgressBar();
+
+            MaterialPostprocessor.s_NeedsSavingAssets = true;
+        }
+
+        [InitializeOnLoadMethod]
+        static void RegisterUpgraderReimport()
+        {
+        }
+    }
+
+    class MaterialPostprocessor : AssetPostprocessor
+    {
+        public static List<string> s_CreatedAssets = new List<string>();
+        internal static List<string> s_ImportedAssetThatNeedSaving = new List<string>();
+        internal static bool s_NeedsSavingAssets = false;
+
+        internal static readonly Action<Material, ShaderID>[] k_Upgraders = {};
+
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
+        {
+            var upgradeLog = "BuiltInRP Material log:";
+            var upgradeCount = 0;
+
+            foreach (var asset in importedAssets)
+            {
+                if (!asset.EndsWith(".mat", StringComparison.InvariantCultureIgnoreCase))
+                    continue;
+
+                var material = (Material)AssetDatabase.LoadAssetAtPath(asset, typeof(Material));
+
+                ShaderID id = ShaderUtils.GetShaderID(material.shader);
+                var wasUpgraded = false;
+
+                var debug = "\n" + material.name;
```

Comment:
Should we be using the explicit newline string here, or something from the generator?

Reply:
I guess that's what the URP version does, so it should be fine?

Reply:
Yeah, this is roughly a straight copy from URP